AdobeStockAsset/Model/Components/SaveCreatorAssetComponent.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeStockAsset\Model\Components;
+
+use Magento\AdobeStockAssetApi\Api\CreatorRepositoryInterface;
+use Magento\AdobeStockAssetApi\Api\Data\CategorySearchResultsInterface;
+use Magento\AdobeStockAssetApi\Api\Data\CreatorInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Exception\AlreadyExistsException;
+
+/**
+ * Class SaveCreatorAssetComponent
+ */
+class SaveCreatorAssetComponent
```

Comment:
These `Save*Component` classes are unnecessary. The repository should be completely and fully responsible for this operation

Reply:
Let me describe the idea of having this approach: if the asset or a category save action would be a simple and straight forward operation I would totally agree with you that their repositories can handle such responsibility. But we have 2 scenarios: 

**The first one:**
for example, category object. The adobe_id is not unique and, as a result, every time you use the `categoryRepository->save($category)` you generate new entry with the same `adobe_id` and possible the same name. As I can see, we do not have the possibility to build one-to-many relation between asset and categories. So this approach is handled inside the save component class.

**The second one**
I decided to propose those services to exclude saving logic from repositories and make them as simple as possible to not make any hidden knowledge. As I can see we will have 2 different types of assets: preview image and licensed. They are similar in some parts but will have different business logic. Having this save services for assets components we can delegate any kind of validation, manipulation and business logic to services and keep repositories with the simplest responsibility - provide data to the presentation layer and save it to domain.